src/tests/multihost/admultidomain/test_multiforest.py

```diff
@@ -11,61 +11,58 @@
           2. Join client to the first domain
           3. Update sssd.conf for second domain
           4. Update krb5.conf for second domain
-          5. Create krb principal and update sssd.conf
+          5. Create krb principal and keytab for second domain in the second forest and update sssd.conf
         :steps:
-          1. Lookup user in the first domain
-          2. Lookup user in the second domain
+          1. Lookup user in the root domain of the first forest
+          2. Lookup user in the  root domain fo the second forest
         :expectedresults:
-          1. User is found in the first domain
-          2. User is found in the second domain
+          1. User is found in the root domain in the first forest
+          2. User is found in the root domain in the second forest
         :customerscenario: True
         """
         adjoin(membersw='adcli')
         ad_domain = multihost.ad[0].domainname
         ad_server = multihost.ad[0].hostname
-        # This must be the last AD server in the metadata file
-        ad1_domain = multihost.ad[len(multihost.ad) - 1].domainname
+        # The second forest domain must be the last entry in the metadata file
+        ad1_ip = multihost.ad[len(multihost.ad)-1].ip
+        ad1_domain = multihost.ad[len(multihost.ad)-1].domainname
         ad1_domain_upper = str.upper(ad1_domain)
-        ad1_server = multihost.ad[len(multihost.ad) - 1].hostname
-        ad1_password = multihost.ad[len(multihost.ad) - 1].ssh_password
+        ad1_server = multihost.ad[len(multihost.ad)-1].hostname
+        ad1_password = multihost.ad[len(multihost.ad)-1].ssh_password
 
-        get_keytab = f'adcli join --host-keytab=/etc/krb5.keytab-domain1 ' \
-                     f'{ad1_domain}'
+        multihost.client[0].run_command('cp -rf /etc/hosts /etc/hosts.bak')
+        multihost.client[0].run_command(f'echo "{ad1_ip}    {ad1_server}" >>  /etc/hosts')
+
+        get_keytab = f'adcli join -v -K /etc/krb5.keytab-domain1 -D {ad1_domain} -S {ad1_server} --stdin-password'
         change_context = 'chcon -t krb5_keytab_t /etc/krb5.keytab-domain1'
         backup_krb5 = 'cp -rf /etc/krb5.conf /etc/krb5.conf.bak'
-        restore_krb5 = 'mv /etc/krb5.conf.bak /etc/krb5.conf ; ' \
-                       'restorecon -Rv /etc/krb5.conf'
+        restore_krb5 = 'mv /etc/krb5.conf.bak /etc/krb5.conf ; restorecon -Rv /etc/krb5.conf'
         cleanup_krb5 = 'rm -rf /etc/krb5.keytab-domain1'
-        edit_krb5_conf = f'sed -i "/domain_realm/a .{ad1_domain} ' \
-                         f'= {ad1_domain_upper}" /etc/krb5.conf'
-        edit1_krb5_conf = f'sed -i "/domain_realm/a {ad1_domain} ' \
-                          f'= {ad1_domain_upper}" /etc/krb5.conf'
-
+        edit_krb5_conf = f'sed -i "/domain_realm/a .{ad1_domain} = {ad1_domain_upper}" /etc/krb5.conf'
+        edit1_krb5_conf = f'sed -i "/domain_realm/a {ad1_domain} = {ad1_domain_upper}" /etc/krb5.conf'
```

Comment:
I know this was just a formatting change, but, in the future, instead of sed's we should use regex or str.replace() to accomplish the same.

```
content = multihost.client[0].get_file_contents
substr = 'domain_realm]\n'
new_lines = f"""\
{ad1_domain} = {ad1_domain_upper}
.{ad1_domain} = {ad1_domain_upper}\
"""
new_content = content.replace(substr, substr + new_lines)
multihost.client[0].put_file_contents('/etc/krb5.conf', new_content)

```